package/MDAnalysis/core/groups.py

```diff
@@ -1876,6 +1954,67 @@ def __getattr__(self, attr):
         raise AttributeError("{cls} has no attribute {attr}"
                              "".format(cls=self.__class__.__name__, attr=attr))
 
+class UpdatingAtomGroup(AtomGroup):
+    """:class:`AtomGroup` subclass that dynamically updates its selected atoms.
+
+    Accessing any attribute/method of an :class:`UpdatingAtomGroup` instance
+    triggers a check for the last frame the group was updated. If the last
+    frame matches the current trajectory frame, the attribute is returned
+    normally; otherwise the group is updated (the stored selections are
+    re-applied), and only then is the attribute returned.
+
+    .. versionadded:: 0.16.0
+
+    """
+
+    def __new__(cls, *args, **kwargs):
+        # If we're reinstantiated in a non-updating way, we must become a
+        # regular AtomGroup. We check to see if we have an AtomGroup passed as
+        # either the 'base_group' kwarg, or the first arg, in which case we
+        # become an UpdatingAtomGroup.
+        bg = kwargs.get('base_group', args[0] if args else None)
+        if isinstance(bg, AtomGroup):
+            return super(UpdatingAtomGroup, cls).__new__(cls, *args, **kwargs)
+        return AtomGroup(*args, **kwargs)
+
+    def __init__(self, base_group, selections):
+        """
+
+        Parameters
+        ----------
+        base_group : :class:`AtomGroup`
+            group on which *selections* are to be applied.
+        selections : a tuple of :class:`~MDAnalysis.core.selection.Selection` instances
+            selections ready to be applied to *base_group*.
+
+        """
+        # Because we're implementing __getattribute__, which needs _u for
+        # its check, no self.attribute access can be made before this line
+        self._u = base_group._u
+        self._base_group = base_group
+        self._updating_selections = selections
+        self._lastupdate = None
+        self.update_selection()
+
+    def update_selection(self):
+        bg = self._base_group
+        sels = self._updating_selections
+        ix = sum([sel.apply(bg) for sel in sels])._ix
+        super(UpdatingAtomGroup, self).__init__(ix, bg._u)
```

Comment:
Ok, no it won't do.
We need the `__new__` for stuff like slicing operations, which internally trigger a call to
`self.__class__(ix, u)`. The `__new__` is needed to divert these cases to a different class altogether (`AtomGroup`).

As for the `super` call, it can be done away with. The only problem might be future changes to `GroupBase.__init__`, that might make its initialization less simple. I already notice that one should also set `self._cache` to the empty dict, and this should be copied into `UpdatingAtomGroup` too. `super` avoids these duplications.
Any other suggestions on how to simplify?